bin/auth0_ls.js

```diff
@@ -0,0 +1,128 @@
+var Chalk = require('chalk');
+var Cli = require('structured-cli');
+var _ = require('lodash');
+var keyValList2Object = require('../lib/keyValList2Object');
+var auth0Extensions = require('./auth0_extensions');
+var extensionTypes = Object.keys(auth0Extensions).sort();
+
+
+module.exports = Cli.createCommand('ls', {
+    description: 'List Auth0 hooks',
+    plugins: [
+        require('./_plugins/profile'),
+    ],
+    handler: handleListAuth0Extensions,
+    optionGroups: {
+        'Hook selection': {
+            'type': {
+                alias: 't',
+                description: 'Hook type. One of: ' + extensionTypes.join(', ') + '.',
+                choices: extensionTypes,
+                dest: 'extensionName',
+                metavar: 'TYPE',
+                type: 'string'
+            },
+        },
+        'Pagination': {
+            'offset': {
+                type: 'int',
+                description: 'Skip this many hooks',
+                defaultValue: 0,
+            },
+            'limit': {
+                type: 'int',
+                description: 'Limit the results to this many hooks',
+                defaultValue: 10,
+            },
+        },
+        'Output options': {
+            'output': {
+                alias: 'o',
+                description: 'Set the output format',
+                choices: ['json'],
+                type: 'string',
+            }
+        },
+    },
+});
+
+
+// Command handler
+
+function handleListAuth0Extensions(args) { 
+    var profile = args.profile;
+
+    keyValList2Object(args, 'meta');
+
+    var meta = {
+        'auth0-extension': 'runtime'
+    };
+    if (args.extensionName)
+        meta['auth0-extension-name'] = args.extensionName;
+    
+    return profile.listWebtasks({ 
+        offset: args.offset, 
+        limit: args.limit, 
+        meta: meta
+    })
+        .catch(function (err) {
+            if (err.statusCode >= 500) throw Cli.error.serverError(err.message);
+            if (err.statusCode >= 400) throw Cli.error.badRequest(err.message);
+            
+            throw err;
+        })
+        .then(onWebtasks);
+        
+    
+    function onWebtasks(webtasks) {
+        var types = {};
+        var count = 0;
+        _.forEach(webtasks, function (webtask) {
+            count++;
+            var json = webtask.toJSON();
+            var record = {
+                name: json.name,
+                type: args.extensionName || (webtask.meta && webtask.meta['auth0-extension-name']) || 'N/A',
+                enabled: !!(webtask.meta && !webtask.meta['auth0-extension-disabled'])
+            };                        
+            if (!types[record.type]) {
+                types[record.type] = [ record ];
+            }
+            else {
+                types[record.type].push(record);
+            }
+        });
+
+        if (args.output === 'json') {
+            console.log(JSON.stringify(types, null, 2));
```

Comment:
Seems counter-intuitive to emit an object that is grouped by extension type in a paginated operation.